lib/jsdom/living/domparsing/parse5-adapter-serialization.js

```diff
@@ -57,3 +57,7 @@ exports.setNodeSourceCodeLocation = (node, location) => {
 };
 
 exports.getNodeSourceCodeLocation = node => node.sourceCodeLocation;
+
+exports.updateNodeSourceCodeLocation = (node, endLocation) => {
+  node.sourceCodeLocation = Object.assign(node.sourceCodeLocation, endLocation);
```

Comment:
`Object.assign` mutates the first argument, so the re‑assignment is unnecessary:
```suggestion
  Object.assign(node.sourceCodeLocation, endLocation);
```